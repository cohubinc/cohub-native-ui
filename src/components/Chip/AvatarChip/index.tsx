import React from "react";
import Avatar from "src/components/Avatar";
import Chip from "../";
import Typography from "src/components/Typography";
import { TChipProps } from "../";
import { View } from "react-native";

interface IAvatarChipProps extends TChipProps {
  name: string;
  avatarUrl: string | null | undefined;
}

export default function AvatarChip(props: IAvatarChipProps) {
  const { name, avatarUrl, ...rest } = props;

  return (
    <Chip
      style={{
        paddingTop: "4px",
        paddingBottom: "4px",
        marginRight: "4px",
        marginBottom: "1rem"
      }}
      dark
      {...rest}
    >
      <View style={{ display: "flex", alignItems: "center" }}>
        <Avatar size={20} src={avatarUrl} className="mr-05" />
        <Typography.Small>{name}</Typography.Small>
      </View>
    </Chip>
  );
}
