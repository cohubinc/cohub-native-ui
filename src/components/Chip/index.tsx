import React, { MouseEventHandler, Component } from "react";

import Color, { ContrastColor } from "src/definitions/enums/Color";
import Typography from "src/components/Typography";
import Icon, { IIconProps } from "src/components/Icon";
import HtmlElementProps from "src/definitions/types/HtmlElementProps";

import AddChipInput from "./AddChipInput";
import AvatarChip from "./AvatarChip";

import "./Chip.scss";

interface IChipProps {
  label?: string;
  onDelete?: MouseEventHandler<HTMLElement>;
  checked?: boolean;
  dark?: boolean;
  backgroundColor?: Color;
  size?: number;
  active?: boolean;
}

export type TChipProps = IChipProps &
  Pick<HtmlElementProps, "onClick" | "className" | "style">;

export default class Chip extends Component<TChipProps> {
  static Add = AddChipInput;
  static Avatar = AvatarChip;

  static defaultProps: Partial<TChipProps> = {
    size: 12,
    backgroundColor: Color.grey300,
    active: false
  };

  render() {
    const {
      children,
      label,
      onClick,
      onDelete,
      checked,
      className = "",
      backgroundColor,
      style,
      size,
      active
    } = this.props;

    const name = label || children;
    const clickable = !!onClick;

    const clickableClass = clickable ? "clickable" : "";

    let iconName: IIconProps["name"] | undefined;
    if (checked) {
      iconName = "checkmark";
    } else if (onDelete) {
      iconName = "close";
    }

    const padding = `${size! / 2.5}px ${size}px`;

    const setBackgroundColor = () => {
      if (active) {
        return Color.green500;
      } else {
        return backgroundColor as Color;
      }
    };

    return (
      <div
        className={`CohubChip ${clickableClass} ${className}`}
        style={{
          backgroundColor: setBackgroundColor() as any,
          borderRadius: "361px",
          display: "inline-block",
          padding,
          ...style
        }}
        onClick={onClick}
        tabIndex={clickable ? 0 : undefined}
      >
        <div
          className="flex justify-center items-center h-100"
          style={{ cursor: clickable ? "pointer" : "inherit" }}
        >
          <Typography.Small color={ContrastColor[setBackgroundColor()] as any}>
            {name}
          </Typography.Small>

          {iconName && (
            <Icon
              onPress={e => onDelete && onDelete(e as any)}
              size={16}
              name={iconName}
              color={ContrastColor[setBackgroundColor()] as any}
            />
          )}
        </div>
      </div>
    );
  }
}
