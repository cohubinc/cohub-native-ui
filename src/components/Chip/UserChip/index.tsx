import React from "react";
import Chip from "..";
import { IUser } from "src/components/graphql";
import Avatar from "src/components/Avatar";
import Typography from "src/components/Typography";
import { TProps as ChipProps } from "..";

interface IProps extends ChipProps {
  user: IUser;
}

export default class UserChip extends React.PureComponent<IProps> {
  render() {
    const { user, ...rest } = this.props;

    return (
      <Chip
        key={user.id}
        style={{
          paddingTop: "4px",
          paddingBottom: "4px",
          marginRight: "4px"
        }}
        dark
        className="mb-1"
        {...rest}
      >
        <div className="flex items-center">
          <Avatar
            size={20}
            src={user.avatar && user.avatar.small_url}
            className="mr-05"
          />
          <Typography.Small>{user.name}</Typography.Small>
        </div>
      </Chip>
    );
  }
}
