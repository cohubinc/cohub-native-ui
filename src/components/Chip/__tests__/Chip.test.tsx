import React from "react";
import { shallow } from "enzyme";

import Chip from "../index";
it("renders internal Link without crashing", async () => {
  const wrapper = shallow(<Chip>Hi, my name is chip!</Chip>);

  expect(wrapper).toMatchSnapshot();
});
