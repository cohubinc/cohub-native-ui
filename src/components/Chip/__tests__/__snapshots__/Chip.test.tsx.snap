// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders internal Link without crashing 1`] = `
<div
  className="CohubChip  "
  style={
    Object {
      "backgroundColor": "hsl(0, 0%, 96%)",
      "borderRadius": "361px",
      "display": "inline-block",
      "padding": "6px 12px",
    }
  }
>
  <div
    className="flex justify-center items-center h-100"
    style={
      Object {
        "cursor": "inherit",
      }
    }
  >
    <Component>
      Hi, my name is chip!
    </Component>
  </div>
</div>
`;
